src/python/lib/ensembl/compara/citest/testdb.py

```diff
@@ -0,0 +1,194 @@
+        table (str): Table to be tested.
+
+    """
+    def __init__(self, name: str, parent: pytest.Item, ref_db: DBConnection, target_db: DBConnection,
+                 table: str, args: Dict) -> None:
+        super().__init__(name, parent, args)
+        self.ref_db = ref_db
+        self.target_db = target_db
+        self.table = table
+
+    def repr_failure(self, excinfo: ExceptionInfo, style: str = None
+                    ) -> Union[str, ReprExceptionInfo, ExceptionChainRepr, FixtureLookupErrorRepr]:
+        """Returns the failure representation that will be displayed in the report section.
+
+        Note:
+            This method is called when ``self.runtest()`` raises an exception.
+
+        Args:
+            excinfo: Exception information with additional support for navigating and traceback.
+            style: Traceback print mode (``auto``/``long``/``short``/``line``/``native``/``no``).
+
+        """
+        if isinstance(excinfo.value, FailedDBTestException):
+            self.error_info['expected'] = excinfo.value.args[0]
+            self.error_info['found'] = excinfo.value.args[1]
+            self.error_info['query'] = str(excinfo.value.args[2]).replace('\n', '').strip()
+            return excinfo.value.args[3] + "\n"
+        if isinstance(excinfo.value, AssertionError):
+            return excinfo.value.args[0] + "\n"
+        return super().repr_failure(excinfo, style)
+
+    def get_report_header(self) -> str:
+        """Returns the header to display in the error report."""
+        return f"Database table: {self.table}, test: {self.name}"
+
+    def test_num_rows(self, variation: float = 0.0, group_by: Union[str, List] = None,
+                      filter_by: Union[str, List] = None) -> None:
+        """Compares the number of rows between reference and target tables.
+
+        If `group_by` is provided, the number of rows will be compared per group, applying the same variation
+        to all of them. If `filter_by` is provided, only the rows matching all the given conditions will be
+        compared.
+
+        Args:
+            variation: Allowed variation between reference and target tables.
+            group_by: Group rows by column(s), and count the number of rows per group.
+            filter_by: Filter rows by one or more conditions (joined by the AND operator).
+
+        Raise:
+            FailedDBTestException: If `group_by` is provided and the number of groups is different; or if the
+                number of rows differ for at least one group.
+
+        """
+        # Compose the sql query from the given parameters (both databases should have the same table schema)
+        table = self.ref_db.tables[self.table]
+        columns = [table.columns[col] for col in to_list(group_by)]
+        # Use primary key in count to improve the query performance
+        primary_key = self.ref_db.get_primary_key_columns(self.table)[0]
```

Comment:
What if the primary key is a compound one ? e.g. `(family_id,seq_member_id)` in `family_member`

Reply:
That is a good question, and I have no idea. I guess that maybe internally MySQL has assigned a new name to that key and that is the one returned and used?